noise_raster/raster_processing.py

```diff
@@ -209,8 +209,8 @@ def vectorize(in_ds, out_poly, selectedTableIndex):
     alg_params = {
         'DATA_TYPE': 5,
         'INPUT_RASTER': in_ds,
-        'NODATA_FOR_MISSING': False,
-        'NO_DATA': -99.0,
+        'NODATA_FOR_MISSING': True,
+        'NO_DATA': 0,
```

Comment:
I checked the PyQGIS function `processing` on the internet. In this commit you are modifying two parameters of its method `run` and I guess that your aim is to set all missing values to 0. Is there a reason for choosing 0 instead of -9999 or -99.0? I am not that familiar with PyQGIS and so a bit out of context.

Reply:
Hi Flaminia,
A raster mask is a raster file with typically two values, 0 and 1. The cells with the value 1 will be kept and the cells with the value 0 will not be kept. In this case, the customer does not want all the values carried over into the shapefile polygon that gets created. They want values that do not exist in their reclassfied tables to be deleted. In order to remove these values, I set the no data value to 0 for the reclassified raster and then used the raster itself as the raster mask (instead of creating a separate raster mask file). You see this a bit further down when I call Polygonize:
```
# Vectorize reclassified raster to create polygon noise contours
    gdal.Polygonize(srcBand=band1, maskBand=band1, outLayer=dst_layer, iPixValField=0)
```
    
Setting the maskBand parameter to the raster itself removes the no data values identified as 0 from the polygon that is created.

Reply:
Thank you, that's clear now.

Reply:
Would it be meaningful to make`selectedTable` not hard-coded?

Reply:
> Would it be meaningful to make`selectedTable` not hard-coded?

The selected table values are required input for the "Reclassify by Table" operation. The user selects "Lden" or "Lnight" in the user interface of the tool. "Lden" categories are used for daytime noise and "Lnight" are used for night time noise. These value ranges are standards used in noise mapping. The input raster is reclassified so that each raster cell gets the same value if it's cell value falls into one of the given ranges. This pre-processing step dramatically improves the performance of the vectorization process. 

What would be the value add of making `selectedTable` not hard-coded?

Reply:
@Kate-Lyndegaard Your explanation why it makes sense to hard-code the table values would make a valuable addition as a comment in the code in my opinion. 

I'd actually suggest be to define the table as a constant in the beginning of the file with your explanation attached and then use the constant here.